app/gui/view/debug_scene/cloud-dashboard/cloud-http/src/lib.rs

```diff
@@ -99,12 +100,46 @@ impl Client {
     /// Returns the list of [`Project`]s the user has access to.
     ///
     /// [`Project`]: ::enso_cloud_view::project::Project
-    pub async fn list_projects(&self) -> Result<response::project::ListProjects, Error> {
+    pub async fn list_projects(&self) -> Result<model::response::project::ListProjects, Error> {
         let route = Route::ListProjects;
         let response = self.try_request(route).await?;
         let projects = response.json().await?;
         Ok(projects)
     }
+
+    /// Changes the [`Project`]s state from [`Closed`] (or [`Created`]) to [`OpenInProgress`] (and
+    /// eventually [`Opened`], once the instance creation completes).
+    ///
+    /// [`Project`]: ::enso_cloud_view::project::Project
+    /// [`Closed`]: ::enso_cloud_view::project::StateTag::Closed
+    /// [`Created`]: ::enso_cloud_view::project::StateTag::Created
+    /// [`OpenInProgress`]: ::enso_cloud_view::project::StateTag::OpenInProgress
+    /// [`Opened`]: ::enso_cloud_view::project::StateTag::Opened
+    pub async fn open_project(
+        &self,
+        project_id: view::id::ProjectId,
+    ) -> Result<model::response::project::OpenProject, Error> {
+        let route = Route::OpenProject(project_id);
+        let response = self.try_request(route).await?;
+        let response = response.json().await?;
+        Ok(response)
```

Comment:
I don't like how these methods are duplicating their code. If feasible, I'd consider if e.g. some trait could be introduced for route types that could simplify this. Or perhaps some helpers could be generated with the macro rules?

Reply:
Right now it's duplicate but this would not be the case for all routes. Some routes are more complex than a plain JSON parse (e.g., we might be interested in the assigned headers). The `try_request` is the bit that extracts the majority of the current common functionality. Though I suppose it can't hurt to move the JSON deserialization into there for now and then refactor as necessary later.